internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package postgres_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/virtualendpoints"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type PostgresqlFlexibleServerVirtualEndpointResource struct{}
+
+func TestAccPostgresqlFlexibleServerVirtualEndpoint_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_virtual_endpoint", "test")
+	r := PostgresqlFlexibleServerVirtualEndpointResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("source_server_id").Exists(),
+				check.That(data.ResourceName).Key("replica_server_id").Exists(),
+				check.That(data.ResourceName).Key("type").Exists(),
+			),
+		},
+		data.ImportStep("type"),
+		data.DisappearsStep(acceptance.DisappearsStepData{
+			Config:       r.basic,
+			TestResource: r,
+		}),
+	})
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := virtualendpoints.ParseVirtualEndpointID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Postgres.VirtualEndpointClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading Postgresql Virtual Endpoint (%s): %+v", id.String(), err)
+	}
+
+	return utils.Bool(resp.Model != nil && resp.Model.Properties != nil && resp.Model.Properties.Members != nil), nil
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Destroy(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := virtualendpoints.ParseVirtualEndpointID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := postgres.DeletePostgresFlexibileServerVirtualEndpoint(ctx, client.Postgres.VirtualEndpointClient, id); err != nil {
+		return nil, fmt.Errorf("deleting Postgresql Virtual Endpoint (%s): %+v", id.String(), err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (PostgresqlFlexibleServerVirtualEndpointResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
```

Comment:
Is there a specific reason we need to add this here? This usually implies that there's an issue with the delete (which we know is the case here) Is the custom poller added not sufficient to workaround it reliably?

Reply:
It was left-over from my local integration tests. I didn't see any harm in leaving it, but it can certainly be removed.